playground/backend/internal/fs_tool/fs.go

```diff
@@ -0,0 +1,113 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fs_tool
+
+import (
+	pb "beam.apache.org/playground/backend/internal/api"
+	"fmt"
+	"github.com/google/uuid"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+const parentBaseFileFolder = "internal"
+
+// Folder contains  names of folders with executable and compiled files (/src and /bin for java SDK)
+type Folder struct {
+	ExecutableFolder string
+	CompiledFolder   string
+}
+
+// Extension contains executable and compiled files' extensions (.java and .class for java SDK)
+type Extension struct {
+	ExecutableExtension string
+	CompiledExtension   string
+}
+
+type LifeCycle struct {
+	folderGlobs *[]string
+	Folder      Folder
+	Extension   Extension
+	pipelineId  uuid.UUID
+}
+
+// NewLifeCycle returns a corresponding LifeCycle depending on the given SDK.
+func NewLifeCycle(sdk pb.Sdk, pipelineId uuid.UUID) (*LifeCycle, error) {
+	switch sdk {
+	case pb.Sdk_SDK_JAVA:
+		return newJavaLifeCycle(pipelineId), nil
+	default:
+		return nil, fmt.Errorf("%s isn't supported now", sdk)
+	}
+}
+
+// CreateFolders creates all folder by folderGlobs.
+func (l *LifeCycle) CreateFolders() error {
+	for _, folder := range *l.folderGlobs {
+		err := os.MkdirAll(folder, fs.ModePerm)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// DeleteFolders deletes all folder by folderGlobs.
```

Comment:
I imagine someone reading this in a web godoc would not know what `folderGlobs` is because it would be hidden.

```
// DeleteFolders deletes all previously provisioned folders
```

Reply:
Done.